marklogic-data-hub/src/main/java/com/marklogic/hub/deploy/commands/LoadUserArtifactsCommand.java

```diff
@@ -322,19 +322,23 @@ private void loadStepDefinitions(HubClient hubClient) {
             for (File typeDir : stepDefsPath.toFile().listFiles(File::isDirectory)) {
                 final String stepDefType = typeDir.getName();
                 for (File defDir : typeDir.listFiles(File::isDirectory)) {
-                    File stepDefFile = new File(defDir, defDir.getName() + ".step.json");
-                    if (stepDefFile.exists()) {
-                        JsonNode stepDef = readArtifact(stepDefFile);
-                        if (!stepDef.has("name")) {
-                            throw new RuntimeException("Unable to load step definition from file: " + stepDefFile +
-                                "; no 'name' property was found");
+                    String[] fileNames;
+                    fileNames = defDir.list();
+                    for (String stepDefFileName : fileNames) {
+                        File stepDefFile = new File(defDir, stepDefFileName);
+                        if (stepDefFile.exists()) {
+                            JsonNode stepDef = readArtifact(stepDefFile);
+                            if (!stepDef.has("name")) {
+                                throw new RuntimeException("Unable to load step definition from file: " + stepDefFile +
+                                    "; no 'name' property was found");
+                            }
+                            final String stepDefName = stepDef.get("name").asText();
+                            logger.info(format("Loading step definition with type '%s' and name '%s'", stepDefType, stepDefName));
+                            service.setArtifact("stepDefinition", stepDefName, stepDef, stepDefFileName.replace(".step.json",""));
```

Comment:
If I understood this right, this new parameter needs to be the parent folder. I have the following example:
..\step-definitions\custom\archiveDoc\afile.step.json
and it creates the following json in the database: /step-definitions/custom/**afile**/afile.step.json
Notice the double use of the step name. And it is used where the parent folder name should be.
Shouldn't we create a json named /step-definitions/custom/archiveDoc/afile.step.json ? Or we need to have the same name twice as a rule?

Reply:
As per my understanding afile directory can have multiple filenames may or may not be of same name as directory.
@ryanjdew what will be your views on that

Reply:
@aniamy This change allows for the situation where we have multiple files under the same directory. We don't generate step definitions that way, but in the scenario where a developer may create variations of a step definition configuration file we want to allow that.